service/history/handler.go

```diff
@@ -681,6 +697,10 @@ func (h *handlerImpl) StartWorkflowExecution(
 
 	startRequest := wrappedRequest.StartRequest
 	workflowID := startRequest.GetWorkflowID()
+
+	// TODO: actually rate limit
```

Comment:
I would avoid creating TODOs, since you will implement it. Not sure how this will help.